File: fe_public/quadra_mall_web-main/src/types/Notification/notification.ts
// src/types/Notification/notification.ts
export interface Notification {
  id: number;
  userId: number;
  type: 'ORDER_UPDATE' | 'PROMOTION' | 'PAYMENT_SUCCESS' | 'TRANSFER' | 'MESSAGE';
  title: string;
  message: string;
  referenceId?: number;
  isRead: boolean; // Sử dụng isRead làm trường chính
  createdAt: string;
  priority: 'high' | 'medium' | 'low';
  category: 'ORDER' | 'PROMOTION' | 'SYSTEM' | 'PAYMENT' | 'MESSAGE';
  icon?: string;
  read?: boolean; // Thêm read để ánh xạ từ backend
}

File: be/quadramall_api-main/src/main/java/com/quadra/ecommerce_api/entity/notification/Notification.java
// com.quadra.ecommerce_api.entity.notification.Notification.java
package com.quadra.ecommerce_api.entity.notification;

import com.quadra.ecommerce_api.entity.user.User;
import com.quadra.ecommerce_api.enums.notification.NotificationType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
@Entity
@Table(name = "notifications")
public class Notification {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Enumerated(EnumType.STRING)
    @Column(name = "type", nullable = false)
    private NotificationType type;

    @Column(name = "title", nullable = false)
    private String title;

    @Column(name = "message", nullable = false, columnDefinition = "TEXT")
    private String message;

    @Column(name = "reference_id")
    private Long referenceId;

    @Column(name = "is_read", nullable = false)
    private boolean isRead = false;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Enumerated(EnumType.STRING)
    @Column(name = "priority", nullable = false)
    private Priority priority = Priority.MEDIUM;

    @Enumerated(EnumType.STRING)
    @Column(name = "category", nullable = false)
    private Category category = Category.SYSTEM;

    @Column(name = "icon")
    private String icon; // Icon cho toast (e.g., 🔔, 📦)

    public enum Priority {
        LOW, MEDIUM, HIGH
    }

    public enum Category {
        ORDER, PROMOTION, SYSTEM, PAYMENT, MESSAGE
    }
}